lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,37 @@
+module Semian
+  class AtomicEnum < AtomicInteger #:nodoc:
+    undef :increment_by
+    undef :increment
+
+    module AtomicEnumSharedImplementation
+      def initialize(name, permissions, symbol_list)
+        super(name, permissions)
+        initialize_lookup(symbol_list)
```

Comment:
You could replace this with:

```
@num_to_sym = Hash[symbol_list.each_with_index.to_a].invert
```

Due to the well-named variable I think it's understandable.